src/transaction/duck_transaction_manager.cpp

```diff
@@ -450,6 +450,7 @@ string DuckTransactionManager::Snapshot(ClientContext &context) {
 }
 
 uint64_t DuckTransactionManager::GetSnapshotId(ClientContext &context) {
+	Checkpoint(context, true);
```

Comment:
Feels like a weird side affect when calling a get function.  Can we add a separate checkpoint function and call that before we call `GetSnapshotId` on the Anybase side?

Reply:
we can look at it as follows: whenever we want to get the snapshot id, we would flush any pending changes to disk
before returning the latest id. Ideally we would like to do it in a single duckdb call instead of of two separate calls: checkpoint + get id.

Reply:
Maybe we should add a separate function for Checkpoint which returns the current ID.  That way it's explicitly known if we are checkpointing and getting the snapshot ID or if we are just asking for the latest ID.

Maybe something like: 

```c++
idx_t DuckTransactionManager::CreateSnapshot(ClientContext &context) {
    Checkpoint(context, true);
    return GetSnapshotId(context);
}
```

Reply:
I agree.